components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2838,6 +2842,7 @@ private FederatedAuthenticatorConfig createFederatedAuthenticatorConfig(String f
         authConfig.setName(authenticatorName);
         authConfig.setDisplayName(getDisplayNameOfAuthenticator(authenticatorName));
         authConfig.setEnabled(authenticator.getIsEnabled());
+        authConfig.setDefinedByType(resolveDefinedByType(authenticatorName));
```

Comment:
IMO, we cannot rely just on the fact whether this authenticator is system or not user by comparing authenticator name.
From the API it should denote that this is a user defined authenticator.
Then server should accept it's name, identifier etc.
If API doesn't say so it should be considered as system. And the authenticator name should be validated against the set of system defined authenticators

Reply:
Addressed with commit fb2b6654746dd4dcb9c324fd84a7a6516149ef9c